src/status_im/chat/events.cljs

```diff
@@ -242,12 +252,112 @@
   (fn [{:keys [get-stored-message]} _]
     (when-not (get-stored-message chat-const/move-to-internal-failure-message-id)
       {:dispatch sign-up/move-to-internal-failure-event})))
-(comment
-  (handlers/register-handler-fx
-    :init-chat
-    [(re-frame/inject-cofx :get-stored-messages)]
-    (fn [{:keys [db get-stored-messages]} _]
-      (let [current-chat-id (:current-chat-id db)]
-        {:db (assoc-in [:chats current-chat-id :messages] (get-stored-messages current-chat-id))
-         ;; TODO(janherich): make this dispatch into fn call once commands loading is refactored
-         :dispatch [:load-commands! current-chat-id]}))))
+
+(handlers/register-handler-fx
+  :init-chat
+  [(re-frame/inject-cofx :get-stored-messages)]
+  (fn [{:keys [db get-stored-messages]} _]
+    (let [current-chat-id (:current-chat-id db)]
+      {:db (assoc-in db [:chats current-chat-id :messages] (get-stored-messages current-chat-id))
+       ;; TODO(janherich): make this dispatch into fn call once commands loading is refactored
+       :dispatch [:load-commands! current-chat-id]})))
+
+(defn- jail-init-callback
+  [{:keys [db] :as fx} chat-id]
+  (let [bot-url     (get-in db [:contacts/contacts chat-id :bot-url])
+        was-opened? (get-in db [:chats chat-id :was-opened?])]
+    (assoc fx :call-jail-function {:chat-id  chat-id
+                                   :function :init
+                                   :context  {:from (:accounts/current-account-id db)}})))
+
+(defn preload-chat-data
+  "Takes coeffects map and chat-id, returns effects necessary when navigating to chat"
+  [{:keys [db get-stored-messages]} chat-id]
+  (let [messages (get-in db [:chats chat-id :messages])
+        commands-loaded? (get-in db [:contacts/contacts chat-id :commands-loaded?])]
+    (cond-> {:db (-> db
+                     (assoc :current-chat-id chat-id)
+                     (assoc-in [:chats chat-id :was-opened?] true)
+                     (model/set-chat-ui-props {:validation-messages nil}))
+             :dispatch-n [[:load-requests! chat-id]]}
+      (not commands-loaded?)
+      (update :dispatch-n conj [:load-commands! chat-id #(re-frame/dispatch [::jail-init-callback chat-id])])
+
+      commands-loaded?
+      (jail-init-callback chat-id)
+      ;; TODO(janherich): what's the purpose of the second term in AND ?
+      (and (seq messages)
+           (not= (count messages) 1))
+      (assoc-in [:db :chats chat-id :messages] (get-stored-messages chat-id)))))
+
+(handlers/register-handler-fx
+  ::jail-init-callback
```

Comment:
do you use :: on purpose here ? the event will be status-im.chat.events/jail-init-callback ? Maybe most events could actually be namespaced :chat

Reply:
The event will only ever be used internally in this namespace, that's why using `::` is quite convenient. 
Another thing which I found good practice is to use implicitly namespaced keywords (`::`) for event which are only used internally in that namespace, it's then immediately clear that one doesn't have to look for invocations/dispatch of such event anywhere else in the app.

Reply:
yes it makes sense, and for externally used I think it is also good to have a simplified namespace like chat here but not in the scope of the PR